Automattic/wp-calypso
# Gutenboarding: Add TrainTracks events for domain picker

This PR adds in TrainTracks events in the domain picker in Gutenboarding. It should fire off events when the search results are rendered in the domain picker, when the search results are updated, and an interact event when a user selects one of the domain suggestions.

More info: PCYsg-bor-p2

#### Changes proposed in this Pull Request

* Add in TrainTracks events for `render` and `interact` in the domain picker following PCYsg-bor-p2 and referencing the logic in the [existing domain step in the legacy signup flow](https://github.com/automattic/wp-calypso/blob/f060d8bc807728561ab1e10dde5fcce44bbce910/client/components/domains/domain-registration-suggestion/index.jsx#L78).
* I've made some assumptions about values to send in the render event, trying to keep it as close as possible to the existing behaviour in the domain step, but swapping out signup for `gutenboarding` in the vendor name.
* Install `@types/uuid`

#### Testing instructions



* Go to `/new` and open up the Domain Picker and examine requests in the network inspector in devtools
* When you change the search query, analytics should fire for each search result
* Check that the `railcarId` matches the pattern `uuid-suggestion0` where the final number is incremented based on its position in the search results — this allows the `interact` event to map to the render event.

E.g. in the following screenshot, the highlighted event is `domain_selected` and has the same railcar id as the rendered suggestion

![image](https://user-images.githubusercontent.com/14988353/79430837-de8d8180-800c-11ea-990f-0b472fdcad65.png)

Questions:

* [ ] Does this cover what we need? Are there additional things (or events) we need to add?
* [ ] Is the behaviour here implemented in the way TrainTracks is meant to work? (I've done it slightly differently to how the existing domains step works, because it looks like the existing domains step doesn't use an iterator in the railcar ID which means the interact events won't map to the particular domain selected — here I've made sure it maps correctly, but _is_ this correct?)
* [ ] Do the TrainTracks events fire too frequently, too infrequently, or are they just right? 🐻

Part of #41059
